scopes/component/isolator/capsule-list.ts

```diff
@@ -27,6 +29,38 @@ export default class CapsuleList extends Array<Capsule> {
   getAllComponents(): Component[] {
     return this.map((c) => c.component);
   }
+  // Sort the capsules by their dependencies. The capsules with no dependencies will be first. Returns a new array.
+  async toposort(host: ComponentFactory): Promise<CapsuleList> {
+    const components = this.getAllComponents();
+    const legacyIdStrToComponentId: { [bitIdStr: string]: ComponentID } = {};
+    const graph = new Graph<Component, string>();
+
+    for (const comp of components) {
+      graph.setNode(new Node(comp.id.toString(), comp));
+    }
+
+    for (const comp of components) {
+      const consumerComp = comp.state._consumer as ConsumerComponent;
+      for (const [depType, depIds] of Object.entries(consumerComp.depsIdsGroupedByType)) {
```

Comment:
never use legacy / consumer component to get dependencies when you are on harmony code.
I saw this code is taken from the scope.main.runtime getGraph func.
We should fix it there is well @davidfirst :)
Anyway this entire code should be removed from here so it won't be a problem anymore.